test/components/views/settings/devices/DeviceDetails-test.tsx

```diff
@@ -60,4 +62,12 @@ describe('<DeviceDetails />', () => {
         const { container } = render(getComponent({ device }));
         expect(container).toMatchSnapshot();
     });
+
+    it('disables sign out button while sign out is pending', () => {
+        const device = {
+            ...baseDevice,
+        };
+        const { getByTestId } = render(getComponent({ device, isSigningOut: true }));
+        expect(getByTestId('device-detail-sign-out-cta')).toMatchSnapshot();
```

Comment:
Instead of using snapshots I would suggest

```suggestion
        expect(getByTestId('device-detail-sign-out-cta')).toBeDisabled();
```

Then we only test the disabled state and not the entire button.